modules/nlu/src/views/full/entities/ListEntity.tsx

```diff
@@ -79,6 +82,11 @@ export const ListEntityEditor: React.FC<Props> = props => {
   }
 
   const editOccurrence = (idx: number, occurrence: NLU.EntityDefOccurrence) => {
+    const newSynonym = _.last(occurrence.synonyms)
+    if (occurrence.synonyms.indexOf(newSynonym) !== occurrence.synonyms.length - 1) {
+      return toastFailure('Synonyms duplication is not allowed')
```

Comment:
that wont work if the content is not sorted (here your sort seems to be triggered by the user)

Reply:
It works as long as new synonym is the last one in the list.
But I agree that's cumbersome - refactored that.